### PR TITLE
Add bluetooth device list

### LINKED ISSUE #35: BLE Device List UI
Creating a new screen for data transfer via Bluetooth. 
Adding a redirect to this screen. 
Creating bluetooth device item with connect button.
Creating a list of available Bluetooth devices using mock data.

### PR DESCRIPTION
**Pull request by AlexJustUser**
## Summary

Added a screen with a list of Bluetooth devices.
Devices are represented by mock data.

## Reasons

It is necessary to add Bluetooth functionality and further expand it into applications.

## References

closes #35

### CODE REVIEW

File: app/src/main/kotlin/com/softteco/template/data/bluetooth/BluetoothDevice.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.softteco.template.data.bluetooth
+
+data class BluetoothDevice(
```
Comment (pcfaktor) [inline app/src/main/kotlin/com/softteco/template/data/bluetooth/BluetoothDevice.kt:3]: Could we avoid default values here?

File: app/src/main/kotlin/com/softteco/template/ui/feature/bluetooth/BluetoothScreen.kt [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import com.softteco.template.ui.components.CustomTopAppBar
+import com.softteco.template.ui.components.TextSnackbarContainer
+import com.softteco.template.ui.theme.Dimens.PaddingDefault
+import com.softteco.template.ui.theme.Dimens.PaddingNormal
+
+@Composable
+fun BluetoothScreen(
+    modifier: Modifier = Modifier,
+    viewModel: BluetoothViewModel = hiltViewModel(),
+    onBackClicked: () -> Unit = {}
```
Comment (pcfaktor) [inline app/src/main/kotlin/com/softteco/template/ui/feature/bluetooth/BluetoothScreen.kt:40]: There is no need for a default value here

File: app/src/main/kotlin/com/softteco/template/ui/feature/bluetooth/BluetoothScreen.kt [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        modifier,
+        onBackClicked
+    )
+}
+
+@Composable
+private fun ScreenContent(
+    state: BluetoothViewModel.State,
+    modifier: Modifier = Modifier,
+    onBackClicked: () -> Unit = {}
```
Comment (pcfaktor) [inline app/src/main/kotlin/com/softteco/template/ui/feature/bluetooth/BluetoothScreen.kt:55]: Likewise. Default value

File: app/src/main/kotlin/com/softteco/template/ui/feature/bluetooth/BluetoothViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        )
+    }.stateIn(
+        viewModelScope,
+        SharingStarted.Lazily,
+        State()
+    )
+
+    @Immutable
+    data class State(
+        val bluetoothDevices: List<BluetoothDevice> = listOf(),
```
Comment (pcfaktor) [inline app/src/main/kotlin/com/softteco/template/ui/feature/bluetooth/BluetoothViewModel.kt:44]: You may use `emptyList()`

**Review by pcfaktor — APPROVED** [review]

**Review by anastasiaShpakava — APPROVED** [review]

**Pull request merged**